src/__tests__/App.test.js

```diff
@@ -0,0 +1,18 @@
+import { Provider } from 'react-redux';
+import TestRenderer from 'react-test-renderer';
+import store from '../app/store';
+import Popular from '../components/Popular';
+import { getCrypto } from '../feature/cryptoSlice';
+
+test('Popular component testing', () => {
+  const TREE = TestRenderer.create(
+    <Provider store={store}>
+      <Popular />
+    </Provider>,
+  );
+  expect(TREE).toMatchSnapshot();
+});
+
```

Comment:
- [ ] I encourage you to add more tests for other components and try to test the pure function in the redux code

Reply:
I had added a test for the getCrypto function

Reply:
I have also added the tests for all components. On the redux code I only have one function which gets data from the API and I have tested it